### PR TITLE
Migrate pull-kubernetes-verify to k8s-infra-prow-build

### PR DESCRIPTION
**Pull request by ZhiFeng1993**
Part of: https://github.com/kubernetes/test-infra/issues/18849

I'll watch this job in the next a few days.

### CODE REVIEW

**Comment by spiffxp** [comment]
Beginning to question the value of this test.  But also, good to know.  If every job scheduled at once, we'd run out of the capacity available in 3 * 18 n1-highmem-8's
```
=== RUN   TestK8sInfraProwBuildJobsMustNotExceedTotalCapacity
--- FAIL: TestK8sInfraProwBuildJobsMustNotExceedTotalCapacity (0.03s)
    jobs_test.go:1121: Total cpu limit 443300m greater than expected limit 432
```

**Comment by spiffxp** [comment]
@ZhiFeng1993 
https://github.com/kubernetes/test-infra/blob/e6266514939056bacc66e40ecc7ffacfd2a8504c/config/tests/jobs/jobs_test.go#L1089-L1090

Can you add another commit to this PR and update these to:
```golang
		coreapi.ResourceCPU:    resource.MustParse("576"),    // 3 * 24 * 8 CPUs per n1-highmem-8
		coreapi.ResourceMemory: resource.MustParse("3744Gi"), // 3 * 24 * 52 Gi per n1-highmem-8
```

**Comment by ZhiFeng1993** [comment]
> Can you add another commit to this PR and update these to:

yes. will do

File: config/tests/jobs/jobs_test.go [hunk 1]
```diff
@@ -1086,8 +1086,8 @@ func TestK8sInfraProwBuildJobsMustHavePodQOSGuaranteed(t *testing.T) {
 func TestK8sInfraProwBuildJobsMustNotExceedTotalCapacity(t *testing.T) {
 	// k8s-infra-prow-build pool1 is 3-zonal 6-30 n1-highmem-8's
 	maxLimit := coreapi.ResourceList{
-		coreapi.ResourceCPU:    resource.MustParse("432"),    // 3 * 18 * 8 CPUs per n1-highmem-8
-		coreapi.ResourceMemory: resource.MustParse("2808Gi"), // 3 * 18 * 52 Gi per n1-highmem-8
+		coreapi.ResourceCPU:    resource.MustParse("576"),    // 3 * 18 * 8 CPUs per n1-highmem-8
+		coreapi.ResourceMemory: resource.MustParse("3744Gi"), // 3 * 18 * 52 Gi per n1-highmem-8
```
Comment (spiffxp) [inline config/tests/jobs/jobs_test.go:1090]:
```suggestion
		coreapi.ResourceCPU:    resource.MustParse("576"),    // 3 * 24 * 8 CPUs per n1-highmem-8
		coreapi.ResourceMemory: resource.MustParse("3744Gi"), // 3 * 24 * 52 Gi per n1-highmem-8
```

Comment (ZhiFeng1993) [reply config/tests/jobs/jobs_test.go]: sorry my bad. Updating

**Pull request merged**